pkg/analyzer/analyze.go

```diff
@@ -0,0 +1,665 @@
+  var (
+    err error
+    node *Node
+  )
+  if node, err = g.Vertex(ID); err !=  nil {
+    fmt.Println("Error getting node attribute", err)
+    os.Exit(1)
+  }
+  val, ok := node.Attributes[key]
+
+  if  !ok {
+    return ID
+  }
+  return val
+}
+
+
+func getPkgResponseFromPurl(ctx context.Context, gqlclient graphql.Client, purl string) (*model.PackagesResponse, error) {
+  pkgInput, err := helpers.PurlToPkg(purl)
+  if err != nil {
+    return nil, fmt.Errorf("failed to parse PURL: %v", err)
+  }
+
+  pkgQualifierFilter := []model.PackageQualifierSpec{}
+  for _, qualifier := range pkgInput.Qualifiers {
+    // to prevent https://github.com/golang/go/discussions/56010
+    qualifier := qualifier
+    pkgQualifierFilter = append(pkgQualifierFilter, model.PackageQualifierSpec{
+      Key:   qualifier.Key,
+      Value: &qualifier.Value,
+    })
+  }
+
+  pkgFilter := &model.PkgSpec{
+    Type:       &pkgInput.Type,
+    Namespace:  pkgInput.Namespace,
+    Name:       &pkgInput.Name,
+    Version:    pkgInput.Version,
+    Subpath:    pkgInput.Subpath,
+    Qualifiers: pkgQualifierFilter,
+  }
+  pkgResponse, err := model.Packages(ctx, gqlclient, *pkgFilter)
+  if err != nil {
+    return nil, fmt.Errorf("error querying for package: %v", err)
+  }
+  if len(pkgResponse.Packages) != 1 {
+    return nil, fmt.Errorf("failed to located package based on purl")
+  }
+  return pkgResponse, nil
+}
+
+func FindHasSBOMBy(filter model.HasSBOMSpec, uri, purl, id string, ctx context.Context, gqlclient graphql.Client) (*model.HasSBOMsResponse, error) {
+  var foundHasSBOMPkg *model.HasSBOMsResponse
+  var err error
+  if purl != "" {
+    pkgResponse, err := getPkgResponseFromPurl(ctx, gqlclient, purl)
+    if err != nil {
+      fmt.Printf("getPkgResponseFromPurl - error: %v", err)
+      return nil, err
+    }
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Subject: &model.PackageOrArtifactSpec{Package: &model.PkgSpec{Id: &pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id}},
+      })
+    if err != nil {
+      fmt.Printf("(purl)failed getting hasSBOM with error :%v", err)
+      return nil, err
+    }
+  } else if uri != ""{
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Uri: &uri,})
+    if err != nil {
+      fmt.Printf("(uri)failed getting hasSBOM  with error: %v", err)
+      return nil, err
+    }
+  }else if id != ""{
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Id: &id,})
+    if err != nil {
+      fmt.Printf("(id)failed getting hasSBOM  with error: %v", err)
+      return nil, err
+    }
+  } else {
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, filter)
+    if err != nil {
+      fmt.Printf("(filter)failed getting hasSBOM  with error: %v", err)
+      return nil, err
+    }
+  }
+  return foundHasSBOMPkg, nil
+}
+
+func verifyAnalyzeFlags(slsas, sboms []string,  errSlsa, errSbom error, uri, purl, id bool) {
```

Comment:
All this should be under the file in `cmd/`, shouldn't have any flag validation here in `pkg/`.